packages/in_app_purchase/lib/src/in_app_purchase_connection/product.dart

```diff
@@ -0,0 +1,8 @@
+import 'package:in_app_purchase/store_kit_wrappers.dart';
+import 'package:flutter/foundation.dart';
+
+class Product {
+  Product({@required this.skProduct});
```

Comment:
I don't think this param should be `@required`, since it's not going to be used on Android.

In general I'm a little wary of throwing something on these high level classes that requires users to write `if (platform) then {` type statements and case in their code. Ideally I was thinking that these wrappers should be as sufficient as possible without needing to know the underlying platform. That's not going to be possible 100% of the time, but I think it should be doable here. I get the benefit here of having access to all of the `SkProductWrapper` information instead of having it be lost, but I'm not sure if it's worth it.